### DISCUSSION TITLE
Why does AnalogAudioStream for ESP32-S3 require an input from two ADC channels and not one?

### DISCUSSION THREAD

**Discussion by candorgander** [pschatzmann/arduino-audio-tools#1082] [Q&A] [answered]
When using ESP32-S3 with Arduino-ESP32 V3.0.0Alphav2 and AnalogAudioStream, the AnalogConfigESP32V1.h header is used when compiling the sketch. Why does the input have to be two channels of the ADC and not one? Have I skipped over some documentation or is this a TODO?

**Comment by pschatzmann** [comment, marked as answer]
What makes you think that it has to be 2?
You can set channels=1

**Reply by candorgander** [reply]
Hmm, I can set that. 

What confused me is that, in [AnalogConfigESP32V1](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_analog_config_e_s_p32_v1.html), [adc_channels](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_analog_config_e_s_p32_v1.html#a3f7103207800e4f0cb1d4c541b8f3408) [2] is an array of two channels & they are declared in AnalogConfigESP32V1.h header file. This would require me to provide two channels, one, the ADC channel that I am using and another as a placeholder. Am I right?

**Reply by pschatzmann** [reply]
Right, if you set one channel, only the first element is relevant: The second is ignored

**Reply by candorgander** [reply]
Right. I will follow that. Would it not make it easier if the functions of AnalogAudioStream is the same over different version of ESP32 core? Is there are reason for the difference?

**Reply by pschatzmann** [reply]
Espessif made a new api that was not available before and it is a he'll of complicated!
